framework/src/modules/chain/transport/transport.js

```diff
@@ -215,37 +215,42 @@ class Transport {
 	 */
 
 	/**
-	 * Description of blocks.
-	 *
-	 * @todo Add @param tags
-	 * @todo Add description of the function
+	 * Returns a set of full blocks starting on the ID defined in the payload up to
+	 * the current tip of the chain.
+	 * @param {object} payload
+	 * @param {string} payload.blockID - The ID of the starting block
+	 * @return {Promise<*|Promise<*>>}
 	 */
-	// eslint-disable-next-line consistent-return
-	async blocks(query) {
-		// Get 34 blocks with all data (joins) from provided block id
-		// According to maxium payload of 58150 bytes per block with every transaction being a vote
-		// Discounting maxium compression setting used in middleware
-		// Maximum transport payload = 2000000 bytes
-		if (!query || !query.lastBlockId) {
-			return {
-				success: false,
-				message: 'Invalid lastBlockId requested',
-			};
-		}
+	async getBlocksFromID(payload) {
+		const validationResult = validator.validate(
+			definitions.getBlocksFromIDRequest,
+			payload,
+		);
 
-		try {
-			const data = await this.blocksModule.loadBlocksFromLastBlockId(
-				query.lastBlockId,
-				34, // 1977100 bytes
+		if (validationResult.length) {
+			const err = validationResult;
+			const error = `${err[0].message}: ${err[0].path}`;
+			this.logger.debug(
+				{
+					err: error,
+					req: payload,
+				},
+				'getBlocksFromID request validation failed',
 			);
-			return { blocks: data, success: true };
-		} catch (err) {
-			return {
-				blocks: [],
-				message: err,
-				success: false,
-			};
+			throw new Error(error);
 		}
+
+		// Get 34 blocks with all data (joins) from provided block id
+		// According to maximum payload of 58150 bytes per block with every transaction being a vote
+		// Discounting maximum compression setting used in middleware
+		// Maximum transport payload = 2000000 bytes
```

Comment:
@shuse2 @ishantiw 
I am tagging you here to catch up on what is the current maximum payload size set for our P2P layer as this will directly affect the number of blocks we can return.

Reply:
Right now by default max payload size is `3048576` bytes

Reply:
@ishantiw can you elaborate on why is that? Thanks!

Reply:
Initially, we had `maxPayloadLimit` around `2000000 bytes` but when we were syncing with this limit, there were some packets having size higher than this limit and it was not syncing after that as it was rejected. To support syncing until the current height we decided to increase it to keep it `3048576 bytes`~3mb.

Reply:
Thanks @ishantiw. @shuse2 does this statement `// According to maximum payload of 58150 bytes per block with every transaction being a vote
` still applies?

Reply:
I think since we change the syncing format on this branch, it should be better.
We need to check size of a block with full vote with 25 transactions.

Reply:
After taking with @shuse2 let's leave it to 34 blocks and discuss this in another spike issue